components/BlocksTable/Table.tsx

```diff
@@ -0,0 +1,104 @@
+import { FC } from "react"
+
+import {
+  Box,
+  NAMED_COLORS,
+  Table,
+  TableProps,
+  Tbody,
+  Th,
+  Thead,
+  Tr,
+  useBreakpointValue
+} from '@ironfish/ui-kit'
+import size from "byte-size"
+
+import { Block } from "types"
+import BlockIcon from "icons/BlockIcon"
+import { truncateHash } from "utils/hash"
+import DataRowSmall from "./DataRowSmall"
+import DataRowLarge from "./DataRowLarge"
+import RowItem from "./RowItem"
+
+size.defaultOptions({
+  precision: 2,
+})
+
+interface BlocksTableProps extends TableProps {
+  data?: Block[],
+  isLoading?: boolean,
+}
+
+const BlocksTable: FC<BlocksTableProps> = ({
+  data = null,
+  isLoading,
+  ...rest
+}) => {
+  const RowData = useBreakpointValue({
+    base: DataRowSmall,
+    lg: DataRowLarge
+  })
+
+  return (
+    <Table
+      {...rest}
+      variant="blocks"
+    >
+      <Thead display={{ base: 'none', lg: 'table-header-group' }}>
+        <Tr>
+          <Th>Block Height</Th>
+          <Th>Size</Th>
+          <Th>TXN</Th>
+          <Th>Block Hash</Th>
+          <Th>Timestamp</Th>
```

Comment:
To prevent label duplications I can suggest creating object or array with labels `columnHeaders`.
F.e.:
```
const columnHeaders = { blockHeight: 'Block Height' }
```
then for THead in can look like this
```
Object.values(columnHeaders).map(header => <Th>{header}</Th>)
```
and for the inline columns
label={columnHeaders.blockHeight}
```